carbon-design-system/carbon-addons-iot-react
# 2623 stateful table ability to expand single one row at a time

Closes #2623 

**Summary**
Adds the ability to expand single ONE row at a time in the stateful table. 

**Change List (commits, features, bugs, etc)**

- Added propType to shape with  expandRowsExclusively to the `hasRowExpansion` prop in the Table
- Updated Stateful table to dispatch the additional setting onRowExpanded
- Updated reducer and creator
- Updated stories and docs
- Wrote tests
- Realised that the stories, documentation and code examples for expansion were incorrect so they were fixed

**Acceptance Test (how to verify the PR)**

1. Go to story https://deploy-preview-2975--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-statefultable--simple-stateful-example
2. Select option `{ expandRowsExclusively: true }` in the hasRowExpansion knob
3. Verify that only one row at a time can be expanded
4. Check language, code examples and spelling in the new docs

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Normal hasRowExpansion:true works as before
- All Table stories with hasRowExpansion works as normal



**Things to look for during review**
